jni/include/faiss_methods.h

```diff
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: Apache-2.0
+//
+// The OpenSearch Contributors require contributions made to
+// this file be licensed under the Apache-2.0 license or a
+// compatible open source license.
+//
+// Modifications Copyright OpenSearch Contributors. See
+// GitHub history for details.
+
+#ifndef OPENSEARCH_KNN_FAISS_METHODS_H
+#define OPENSEARCH_KNN_FAISS_METHODS_H
+
+#include "faiss/Index.h"
+#include "faiss/IndexBinary.h"
+#include "faiss/IndexIDMap.h"
+#include "faiss/index_io.h"
+
+namespace knn_jni {
+namespace faiss_wrapper {
+
+/**
+ * A class having wrapped faiss methods
+ *
+ * This class helps to mock faiss methods during unit test
+ */
+class FaissMethods {
```

Comment:
Are we missing some methods? i.e. load, free, search,...

Reply:
I haven't moved all methods yet. We will add incrementally.

Reply:
Lets add todo

Reply:
I didn't want to go through and see what is needed to be in this class to add todo. We can add methods as needed base. For example, we don't need to add search method here because search is a method of Index struct which can be mocked without this class.